File: python/05_policy_block.py
"""Handle a policy block, then retry after a redact-mode profile in the portal.

    export TONIA_API_KEY=tonia_sk_…
    python 05_policy_block.py
"""

from __future__ import annotations

import os
import sys

from tonia import PolicyBlockError, Tonia

with Tonia(api_key=os.environ["TONIA_API_KEY"]) as client:
    try:
        client.chat.completions.create(
            model="openai/gpt-4.1-mini",
            messages=[{"role": "user", "content": "…"}],
        )
    except PolicyBlockError as err:
        print("Blocked by policy.", file=sys.stderr)
        print(
            "Configure a redact-mode profile in the portal, bind this key, then retry.",
            file=sys.stderr,
        )
        print(err.policy_block, file=sys.stderr)
        raise SystemExit(1) from err

File: typescript/05-policy-block.ts
/**
 * Handle a policy block, then retry after a redact-mode profile is configured
 * in the tonia portal (https://portal.tonia.ca).
 *
 *   export TONIA_API_KEY=tonia_sk_…
 *   npx tsx 05-policy-block.ts
 */
import { PolicyBlockError, Tonia } from "@tonia/sdk";

const client = new Tonia({ apiKey: process.env.TONIA_API_KEY });

try {
  await client.chat.completions.create({
    model: "openai/gpt-4.1-mini",
    messages: [{ role: "user", content: "…" }],
  });
} catch (err) {
  if (err instanceof PolicyBlockError) {
    console.error("Blocked by policy.");
    console.error("Configure a redact-mode profile in the portal, bind this key, then retry.");
    console.error(err.policyBlock);
    process.exitCode = 1;
  } else {
    throw err;
  }
}
